src/actions/sources/loadSourceText.js

```diff
@@ -31,17 +40,29 @@ async function loadSource(source: Source, { sourceMaps, client }) {
  */
 export function loadSourceText(source: Source) {
   return async ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
+    console.log("lst", source);
     // Fetch the source text only once.
-    if (source.text) {
+    if (isLoaded(source)) {
       return Promise.resolve(source);
     }
 
+    if (isLoading(source)) {
```

Comment:
this requires that there has been a state update before this happened, however it could be that two requests with the same data will pass this check, thus resulting in the same issue we currently have. it would be better to do 

```javascript
const existingRequest = requests.get(source.id);
if (existingRequest) {
  return existingRequest
}

Reply:
Ugh... sources of truth :)